src/api-v1.js

```diff
@@ -580,6 +580,94 @@ api.declare({
 
 api.declare({
   method: 'put',
+  route: '/ami-set/:amiSetId',
+  name: 'createAmiSet',
+  deferAuth: true,
+  scopes: [['aws-provisioner:manage-ami-set:<amiSetId>']],
+  title: 'Create new AMI Set',
+  stability:  base.API.stability.stable,
+  description: [
+    'Create an AMI Set. An AMI Set is a collection of AMIs with a single name.',
+  ].join('\n'),
+}, async function (req, res) {
+  let input = req.body;
+  let AmiSetId = req.params.id;
+
+  input.lastModified = new Date();
+
+  // Authenticate request with parameterized scope
+  if (!req.satisfies({amiSet: AmiSetId})) {
```

Comment:
the property that is checked against (here `amiSet`) needs to match the parameterized name in the scopes list (here `amiSetId`, but should be `id`).

After renaming the `amiSetId` variables and scopes to `id`, the object to use here would be `{id: id}`.  In es6+, if your property name and value are derived from the save variable, e.g. `let x = 1; {x: x}` then you can just do `let x = 1; {x}.